src/ast/mod.rs

```diff
@@ -201,6 +201,12 @@ pub enum Expr {
         expr: Box<Expr>,
         data_type: DataType,
     },
+    /// TRY_CAST an expression to a different data type e.g. `TRY_CAST(foo AS VARCHAR(123))`
+    //  this differs from CAST in the choice of how to implement invalid conversions
+    TryCast {
```

Comment:
I am wondering, why the choice `TryCast` / `try_cast`?

I know bigquery uses `SAFE_CAST` https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#safe_casting

It would make sense to support both maybe?

Reply:
I could easily change this to add SAFE_CAST too. There was some rationale discussed here: https://github.com/apache/arrow/pull/9682#issuecomment-799831930

basically it was just that I think SQL Server got there first with TRY_CAST